vars/pythonSetupPyPipeline.groovy

```diff
@@ -218,8 +218,14 @@ def call(Map pipelineParams) {
 
           // Deploy the html documents to the docs branch, which when using "BitBucket Pages" allows serving the documentation directly
           stage("Deploy Docs") {
+            agent {
+              docker {
+                image buildImageName
+                args pipelineParams.dockerRunArgs
+                reuseNode true
+              }
+            }
             when {
-              beforeAgent true
```

Comment:
Why did you remove this? This ensures that the below conditions are evaluated first before the agent is actually used.

See https://jenkins.io/doc/book/pipeline/syntax/

> If beforeAgent is set to true, the when condition will be evaluated first, and the agent will only be entered if the when condition evaluates to true.

Reply:
To check if ghp-import plugin is installed in the when condition, the agent has to be used. So, had to remove this beforeAgent being set to true.

Reply:
Can you please add this explanation to the code.